docs/oauth-server.md

```diff
@@ -0,0 +1,144 @@
+
+The following steps must be performed by an administrator of the Google Apps domain:
+
+1.  Go to your Google Apps domain’s [Admin console](http://admin.google.com).
+2.  Select **Security** from the list of controls. If you don't see **Security** listed, select **More controls** from the gray bar at the bottom of the page, then select **Security** from the list of controls. If you can't see the controls, make sure you're signed in as an administrator for the domain.
+3.  Select **Advanced settings** from the list of options.
+4.  Select **Manage third party OAuth Client access** in the **Authentication** section.
+5.  In the **Client name** field enter the service account's **Client ID**.
+6.  In the **One or More API Scopes** field enter the list of scopes that your application should be granted access to. For example, if your application needs domain-wide access to the Google Drive API and the Google Calendar API, enter: https://www.googleapis.com/auth/drive, https://www.googleapis.com/auth/calendar.
+7.  Click **Authorize**.
+
+Your application now has the authority to make API calls as users in your domain (to "impersonate" users). When you prepare to make authorized API calls, you specify the user to impersonate.
+
+[](#top_of_page)Preparing to make an authorized API call
+--------------------------------------------------------
+
+After you have obtained the client email address and private key from the Developers Console, set the path to these credentials in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable ( **Note:** This is not required in the App Engine environment):
+
+```php
+putenv('GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account.json');
+```
+
+Call the `useApplicationDefaultCredentials` to use your service account credentials to authenticate:
+
+```php
+$client = new Google_Client();
+$client->useApplicationDefaultCredentials();
+```
+
+If you have delegated domain-wide access to the service account and you want to impersonate a user account, specify the email address of the user account using the method `setSubject`:
+
+```php
+$client->setSubject($user_to_impersonate);
+```
+
+Use the authorized `Google_Client` object to call Google APIs in your application.
+
+## Calling Google APIs
+
+Use the authorized `Google_Client` object to call Google APIs by completing the following steps:
+
+1.  Build a service object for the API that you want to call, providing the authorized `Google_Client` object. For example, to call the Cloud SQL Administration API:
+    
+    ```php
+    $sqladmin = new Google_Service_SQLAdmin($client);
+    ```
+    
+2.  Make requests to the API service using the [interface provided by the service object](https://devsite.googleplex.com/api-client-library/php/start/get_started#build). For example, to list the instances of Cloud SQL databases in the examinable-example-123 project:
```

Comment:
This should be going [here](https://github.com/googleapis/google-api-php-client/pull/1668/files#diff-e2348c79e582266d6ee3351360bccf3dR62)